stdlib.sh

```diff
@@ -19,7 +19,16 @@ direnv="$(command -v direnv)"
 DIRENV_LOG_FORMAT="${DIRENV_LOG_FORMAT-direnv: %s}"
 
 # Where direnv configuration should be stored
-direnv_config_dir="${DIRENV_CONFIG:-${XDG_CONFIG_HOME:-$HOME/.config}/direnv}"
+IFS=: xdg_config_dirs=( ${XDG_CONFIG_DIRS:-/etc/xdg} )
+xdg_config_dirs+=( "${XDG_CONFIG_HOME:-$HOME/.config}" )
+
+declare -a direnv_config_dirs
+for dir in "${xdg_config_dirs[@]}"; do
+  direnv_config_dirs+=( "$dir/direnv" )
+done
+if [[ -n "$DIRENV_CONFIG" ]]; then
+  direnv_config_dirs+=( "$DIRENV_CONFIG" )
+fi
```

Comment:
@zimbatm Btw this also slightly changes behavior: Before when `DIRENV_CONFIG` was set, `$XDG_CONFIG_HOME`/`~/.config` was ignored, but now it's loaded regardless. I can easily change the behavior to work as before, what do you think?

Reply:
Given how widely direnv is deployed, it's best to keep as much back-compat as possible. DIRENV_CONFIG is for when you want to override the default behaviour.

Reply:
I fixed this in https://github.com/direnv/direnv/pull/990/commits/f106329094d9ad67523e64f6837b1b9b89e36d71 now, it has the old behavior again